tiledb/sm/filesystem/azure.cc

```diff
@@ -137,47 +140,49 @@ Status Azure::init(const Config& config, ThreadPool* const thread_pool) {
 
   write_cache_max_size_ = max_parallel_ops_ * block_list_block_size_;
 
-  // Initialize a credential object
-  shared_ptr<azure::storage_lite::storage_credential> credential =
-      make_shared<azure::storage_lite::shared_key_credential>(
-          HERE(), account_name, account_key);
+  // Initialize logging from the Azure SDK.
+  static std::once_flag azure_log_sentinel;
+  std::call_once(azure_log_sentinel, []() {
+    ::Azure::Core::Diagnostics::Logger::SetListener(
+        [](auto level, const std::string& message) {
+          switch (level) {
+            case ::Azure::Core::Diagnostics::Logger::Level::Error:
+              LOG_ERROR(message);
+              break;
+            case ::Azure::Core::Diagnostics::Logger::Level::Warning:
+              LOG_WARN(message);
+              break;
+            case ::Azure::Core::Diagnostics::Logger::Level::Informational:
+              LOG_INFO(message);
+              break;
+            case ::Azure::Core::Diagnostics::Logger::Level::Verbose:
+              LOG_DEBUG(message);
+              break;
+          }
+        });
+  });
 
-  // Use the SAS token, if provided.
-  if (!sas_token.empty()) {
-    // clang-format off
-    credential = make_shared<azure::storage_lite::shared_access_signature_credential>(HERE(), sas_token);
-    // clang-format on
-  }
-
-  auto account = make_shared<azure::storage_lite::storage_account>(
-      HERE(), account_name, credential, use_https, blob_endpoint);
-
-  // Construct the Azure SDK blob client with a concurrency level
-  // equal to 'thread_pool_->concurrency_level'. Internally, the client
-  // will allocate an equal number of libcurl sessions with
-  // 'curl_easy_init'. This ensures that our 'thread_pool_' threads
-  // will not block on the blob client's internal request queue
-  // unless the user is performing concurrent I/O on this instance.
-
-  // Get CA Cert bundle file from global state. This is initialized and cached
-  // if detected. We have only had issues with finding the certificate path on
-  // Linux.
-  if constexpr (tiledb::platform::PlatformCertFile::enabled) {
-    const std::string cert_file = tiledb::platform::PlatformCertFile::get();
-    client_ = make_shared<azure::storage_lite::blob_client>(
-        HERE(), account, thread_pool_->concurrency_level(), cert_file);
+  ::Azure::Storage::Blobs::BlobClientOptions options;
+  options.Retry.MaxRetries = constants::azure_max_attempts;
```

Comment:
Can we make these config options with defaults? We should also standardize the timing and retry count with what we have for AWS if we can, will just make it simplier to document and explain to users.

Reply:
I added config options. Standardizing across all cloud providers is hard because each SDK allows configuring the retry policies differently and with varying degrees of freedom.